lesson6-Reflection/Task1/Container.cs

```diff
@@ -1,28 +1,100 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Reflection;
+using Task1.DoNotChange;
 
 namespace Task1
 {
     public class Container
     {
+        private readonly Dictionary<Type, Type> _mappedTypes = new Dictionary<Type, Type>();
+
         public void AddAssembly(Assembly assembly)
         {
-            throw new NotImplementedException();
+            var mappedTypes = assembly.GetTypes().Where(t =>
+                Attribute.IsDefined(t, typeof(ExportAttribute)) ||
+                Attribute.IsDefined(t, typeof(ImportConstructorAttribute)) ||
+                t.GetProperties().Any(p => Attribute.IsDefined(p, typeof(ImportAttribute))) ||
+                t.GetFields().Any(f => Attribute.IsDefined(f, typeof(ImportAttribute))));
+
+            foreach (var type in mappedTypes)
+                if (type.GetCustomAttributes(typeof(ExportAttribute)).FirstOrDefault() == null)
```

Comment:
```suggestion
		{
			var contract = (type.GetCustomAttributes(typeof(ExportAttribute)).FirstOrDefault() as ExportAttribute)
				?.Contract;
			if (contract != null)
				AddType(type, contract);
			else
				AddType(type);
		}
```
Reason: reading attributes through reflection is resource consuming, this way we can reduce number of such operations.